jcl/src/java.base/share/classes/jdk/internal/vm/Continuation.java

```diff
@@ -176,7 +176,7 @@ private static void enter(Continuation cont) {
 		try {
 			cont.runnable.run();
 		} finally {
-			cont.finished = true;
+			cont.setFinished(true);
```

Comment:
let's pass the flag as a parameter of yield rather then having explicit setFinished java/native methods, to avoid entering native twice